File: cartracker/src/main/java/ch/rasc/cartracker/entity/option/DriveTrain.java
package ch.rasc.cartracker.entity.option;

import javax.persistence.Entity;

import ch.rasc.extclassgenerator.Model;

@Entity
@Model(value = "CarTracker.model.option.DriveTrain",
		readMethod = "driveTrainOptionService.read",
		createMethod = "driveTrainOptionService.create",
		updateMethod = "driveTrainOptionService.update",
		destroyMethod = "driveTrainOptionService.destroy", paging = true)
public class DriveTrain extends BaseOption {
	// nothing special
}

File: cartracker/src/main/webapp/app/model/option/DriveTrain.js
Ext.define("CarTracker.model.option.DriveTrain",
{
  extend : "Ext.data.Model",
  fields : [ {
    name : "longName",
    type : "string"
  }, {
    name : "shortName",
    type : "string"
  }, {
    name : "createDate",
    type : "date",
    dateFormat : "c"
  }, {
    name : "active",
    type : "boolean",
    defaultValue : true
  }, {
    name : "id",
    type : "int",
    useNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "driveTrainOptionService.read",
      create : "driveTrainOptionService.create",
      update : "driveTrainOptionService.update",
      destroy : "driveTrainOptionService.destroy"
    },
    reader : {
      root : "records"
    }
  }
});